samples/local_attestation/common/attestation.cpp

```diff
@@ -12,154 +19,159 @@ Attestation::Attestation(Crypto* crypto, uint8_t* enclave_mrsigner)
  * the mrsigner, product_id, isvsvn values are checked to  see if they are
  * predefined trusted values.
- * 3) Once the enclave's trust has been established, the validity of
- * accompanying data is ensured by comparing its SHA256 digest against the
- * report_data field.
  */
-bool Attestation::attest_local_report(
-    const uint8_t* local_report,
-    size_t report_size,
+bool Attestation::attest_local_evidence(
+    const uint8_t* local_evidence,
+    size_t evidence_size,
     const uint8_t* data,
     size_t data_size)
 {
     bool ret = false;
     uint8_t sha256[32];
-    oe_report_t parsed_report = {0};
     oe_result_t result = OE_OK;
+    oe_result_t verifier_result = OE_OK;
+    oe_claim_t* claims = NULL;
+    size_t claims_length = 0;
 
-    // While attesting, the report being attested must not be tampered
+    // While attesting, the evidence being attested must not be tampered
     // with. Ensure that it has been copied over to the enclave.
-    if (!oe_is_within_enclave(local_report, report_size))
+    if (!oe_is_within_enclave(local_evidence, evidence_size))
     {
-        TRACE_ENCLAVE("Cannot attest report in host memory. Unsafe.");
+        TRACE_ENCLAVE("Cannot attest evidence in host memory. Unsafe.");
         goto exit;
     }
 
-    TRACE_ENCLAVE("report_size = %ld", report_size);
+    TRACE_ENCLAVE("evidence_size = %ld", evidence_size);
 
-    // 1)  Validate the report's trustworthiness
-    // Verify the report to ensure its authenticity.
-    result = oe_verify_report(local_report, report_size, &parsed_report);
+    verifier_result = oe_verifier_initialize();
+    if (verifier_result != OE_OK)
+    {
+        TRACE_ENCLAVE("oe_verifier_initialize failed.");
+        goto exit;
+    }
+
+    // 1)  Validate the evidence's trustworthiness
+    // Verify the evidence to ensure its authenticity.
+    result = oe_verify_evidence(&sgx_local_uuid, local_evidence, evidence_size, NULL, 0, NULL, 0, &claims, &claims_length);
     if (result != OE_OK)
     {
-        TRACE_ENCLAVE("oe_verify_report failed (%s).\n", oe_result_str(result));
+        TRACE_ENCLAVE("oe_verify_evidence failed (%s).\n", oe_result_str(result));
         goto exit;
     }
 
-    TRACE_ENCLAVE("oe_verify_report succeeded\n");
+    TRACE_ENCLAVE("oe_verify_evidence succeeded\n");
 
-    // 2) validate the enclave identity's signed_id is the hash of the public
-    // signing key that was used to sign an enclave. Check that the enclave was
-    // signed by an trusted entity.
-    if (memcmp(parsed_report.identity.signer_id, m_enclave_mrsigner, 32) != 0)
+    // Iterate through list of claims.
+    for (size_t i = 0; i < claims_length; i++) 
     {
-        TRACE_ENCLAVE("identity.signer_id checking failed.");
-        TRACE_ENCLAVE(
-            "identity.signer_id %s", parsed_report.identity.signer_id);
-
-        for (int i = 0; i < 32; i++)
+        if (strcmp(claims[i].name, OE_CLAIM_SIGNER_ID) == 0)
```

Comment:
I think the claim check implemented [here](https://github.com/openenclave/openenclave/blob/master/tests/attestation_plugin/plugin/tests.c#L361-L372) might be a better choice. You can abstract a function like `find_claim()` to find the claim first then do the check. It's cleaner and more efficient.
Current approach iterates the claim list once but need 3 `strcmp`s for each claim while `find_claim()` perform 3 times of list iteration but will early stop in each iteration.

Reply:
I agree with that approach. I have updated the code to reflect that.